llvm/test/tools/sycl-post-link/exclude_external_functions.ll

```diff
@@ -0,0 +1,67 @@
+; Test that when the -exclude-exernal-funtions option is used,
+; dependencies to SYCL External functions do not cause the SYCL External function
+; to be added to a device image.
+
+; RUN: sycl-post-link -symbols -exclude-external-functions -split=kernel -S < %s -o %t.table
```

Comment:
I think that we should add a separate `per-source` split test that contains calls between `SYCL_EXTERNAL` functions from both different and the same translation units.

In that device code split mode, we don't expect `SYCL_EXTERNAL` functions that belong to the same translation unit to be outlined into different device images regardless of `-exclude-external-functions` flag presence.

Reply:
New test added, exclude_external_functions_source.ll